crates/common/tedge_utils/src/fs_notify.rs

```diff
@@ -0,0 +1,574 @@
+use std::{
+    collections::HashMap,
+    hash::Hash,
+    path::{Path, PathBuf},
+};
+
+use async_stream::try_stream;
+pub use futures::{pin_mut, Stream, StreamExt};
+use inotify::{EventMask, Inotify, WatchMask};
+use std::collections::BTreeSet;
+use strum_macros::Display;
+use try_traits::default::TryDefault;
+
+#[derive(Debug, Display, PartialEq, Eq, Clone, Hash, PartialOrd, Ord, Copy)]
+pub enum FileEvent {
+    Modified,
+    Deleted,
+    Created,
+}
+
+impl From<FileEvent> for WatchMask {
+    fn from(value: FileEvent) -> Self {
+        match value {
+            FileEvent::Modified => WatchMask::MODIFY,
```

Comment:
Is `MODIFY` a mask that includes `CLOSE_WRITE` ?

If not I would expect the `FileEvent::Modified` translated into a mask that includes all the file-update related events.

Reply:
I believe that we should have either CLOSE_WRITE or MODIFIED but not both. 

MODIFIED event is triggered after: 

```c
write(fd, buf, count);
```
CLOSE_WRITE event is triggered after:
```c
close(fd);
```

If we include both, then we get duplicate events. 

Furthermore, if we include CLOSE_WRITE we get false positives (a user simply opening and then closing a file will trigger the event even though nothing was changed.)  This will make our plugins send unchanged payloads of operations/supported log types or supported configuration etc.

I see MODIFY as the better fit for our use case but I can change it to CLOSE_WRITE. But, in no case we should have both. 

Initially, the goal of this API was to simplify options to the user by combining flags, but now I see it more that this API should protect users from silly flag combinations (such as MODIFY and CLOSE_WRITE).

> As a side, note there is a CLOSE_NOWRITE flag but this does something else. It is for when a read-only file was closed. 

This is where I got the information from: https://man7.org/linux/man-pages/man7/inotify.7.html

Reply:
As another side-note, if we wanted a `FileEvent::Closed` event and we did not care if the file is read-only or writable, we could combine: 

```rust
match value {
   FileEvent::Modified => WatchMask::MODIFY,
   FileEvent::Closed => WatchMask::CLOSE_WRITE | WatchMask::CLOSE_NOWRITE,
   ...
}
```